spec/requests/secret_shares_spec.rb

```diff
@@ -1,110 +1,116 @@
 # frozen_string_literal: true
 
-describe SecretSharesController, type: :controller do
+describe 'SecretShares', type: :request do
   def raise_record_not_found
     raise_error(ActiveRecord::RecordNotFound)
   end
 
   describe 'POST create' do
-    def do_post_create
-      post :create, params: { moment: moment }
-    end
     let(:moment) { create(:moment, :with_user) }
+
     context 'signed in as creator of the moment' do
-      before do
-        sign_in moment.user
-        do_post_create
-      end
+      before { sign_in moment.user }
 
       it 'Creates Secret Share Identifier' do
+        post secret_shares_path, params: { moment: moment.slug }
+
         expect(moment.reload.secret_share_identifier).not_to be_nil
         expect(response).to redirect_to moment_path(moment, anchor: 'secretShare')
       end
     end
 
     context 'signed in as another user' do
-      before do
-        new_user = create(:user)
-        sign_in new_user
-      end
+      let(:user) { create(:user) }
 
-      specify { expect { do_post_create }.to raise_record_not_found }
-    end
+      before { sign_in user }
 
-    context 'not signed in' do
-      before do
-        do_post_create
+      it 'returns not found status' do
+        post secret_shares_path, params: { moment: moment.slug }
+
+        expect(response).to have_http_status(:not_found)
       end
+    end
 
+    context 'not signed in' do
       it 'does not create Secret Share Identifier' do
+        post secret_shares_path, params: { moment: moment.slug }
+
         expect(moment.reload.secret_share_identifier).to be_nil
       end
 
       it 'redirects to sign in page' do
+        post secret_shares_path, params: { moment: moment.slug }
+
         expect(response).to redirect_to new_user_session_path
       end
     end
   end
 
   describe 'GET show' do
-    def do_get_show(secret_share_identifier = moment.secret_share_identifier)
-      get :show, params: { id: secret_share_identifier }
-    end
     context 'secret share is valid' do
-      before { do_get_show }
       let(:moment) { create(:moment, :with_user, :with_secret_share) }
-      specify { expect(response).to render_template(:show) }
+
+      it 'renders show template' do
+        get secret_share_path(id: moment.secret_share_identifier)
+
+        expect(response).to render_template('show')
```

Comment:
Let's do this assertion ` expect(response).to be_successful` instead since `render_template` is deprecated.